src/python/qeqiskit/backend/backend.py

```diff
@@ -241,94 +335,52 @@ def aggregregate_measurements(
-            experiments, n_samples_for_experiments
-        )
-
-        jobs = [
-            self.execute_with_retries(batch, n_samples)
-            for n_samples, batch in zip(n_samples_for_batches, batches)
-        ]
-
-        self.number_of_circuits_run += len(circuits)
-        self.number_of_jobs_run += len(batches)
-
-        return self.aggregregate_measurements(jobs, batches, multiplicities)
-
-    def execute_with_retries(
-        self, batch: List[QuantumCircuit], n_samples: int
-    ) -> IBMQJob:
-        """Execute a job, resubmitting if the the backend job limit has been
-        reached.
-
-        The number of seconds between retries is specified by
-        self.retry_delay_seconds. If self.retry_timeout_seconds is defined, then
-        an exception will be raised if the submission does not succeed in the
-        specified number of seconds.
-
-        Args:
-            batch: The batch of qiskit ircuits to be executed.
-            n_samples: The number of shots to perform on each circuit.
+        Raises:
+            TypeError: If n_samples_for_readout_correction was not defined when the
+                QiskitBackend Object was declared.
 
         Returns:
-            The qiskit representation of the submitted job.
+            mitigated_counts (Counts): counts for each output bitstring only showing
+                the qubits which were mitigated.
         """
 
-        start_time = time.time()
-        while True:
-            try:
-                job = execute(
-                    batch,
-                    self.device,
-                    shots=n_samples,
-                    basis_gates=self.basis_gates,
-                    optimization_level=self.optimization_level,
-                    backend_properties=self.device.properties(),
-                )
-                return job
-            except IBMQBackendJobLimitError:
-                if self.retry_timeout_seconds is not None:
-                    elapsed_time_seconds = time.time() - start_time
-                    if elapsed_time_seconds > self.retry_timeout_seconds:
-                        raise RuntimeError(
-                            f"Failed to submit job in {elapsed_time_seconds}s due to "
-                            "backend job limit."
-                        )
-                print(f"Job limit reached. Retrying in {self.retry_delay_seconds}s.")
-                time.sleep(self.retry_delay_seconds)  # type: ignore
+        for key in counts.keys():
+            num_qubits = len(key)
+            break
 
-    def _apply_readout_correction(self, counts, qubit_list=None):
-        if self.readout_correction_filter is None:
+        if active_qubits is None:
+            active_qubits = list(range(num_qubits))
+        else:
+            active_qubits.sort()
+            for key in deepcopy(list(counts.keys())):
+                new_key = "".join(key[i] for i in active_qubits)
```

Comment:
I don't think joining integers without any separators is a good idea, as a key such as: `134`, can mean any of: `1,3,4`, `13, 4`, `1, 34`. 
Also, we would need to have a test case that would catch this.

Reply:
good point! Will fix.

Reply:
Oh wait, the reason i did this is because I am removing the bits corresponding to non active_qubits, not using active_qubits as the keys to the counts dictionary.

Reply:
Either way it seems error prone to me ;)

Reply:
Isn't this just the way that we do counts of bitstrings in `core`? Probably beyond the scope of this PR to fix this, right? So maybe we should make another story?